### ISSUE TITLE
GitRepoProvider seems to redirect to github?

### ISSUE THREAD

**Issue by adl** [jupyterhub/binderhub#475]
Hi,

I have a binder-ready repository which you can clone `https://gitlab.lrde.epita.fr/adl/binder-test.git`, and I would like to try commit `304d0e858f7ad031fc83cf0eb288b954c2d7123e`.

For the [documentation of `class GitRepoProvider'](https://github.com/jupyterhub/binderhub/blob/master/binderhub/repoproviders.py#L114), I understand I should use the following link to launch the binder:

https://mybinder.org/v2/git/https%3A%2F%2Fgitlab.lrde.epita.fr%2Fadl%2Fbinder-test.git/223304d0e858f7ad031fc83cf0eb288b954c2d7123e

However, clicking on this link unexpectedly redirects me to `https://mybinder.org/v2/gh/https://gitlab.lrde.epita.fr/adl/binder-test.git/master` which (unsurprisingly) fails to load.

**Comment by choldgraf** [comment]
ah - so I believe that right now we have support for the build machines to
pull from git repos, but this hasn't been built into the Binder URL API yet
(e.g. the stuff that starts with `v2/git`. I think this will need some
coding before we can share generic git repo URLs
--

**Comment by choldgraf** [comment]
yeah the build form packages the repo info and sends it to the `/build/` endpoint, which starts the build process. The `/build/` endpoint has git repo support (and gitlab) support, but the UI does not yet. If this is something that'd be really useful to you, PRs are super welcome! Otherwise eventually it will happen, just a question of bandwidth :-)
